### PR TITLE
Allow renderParts to be used in pure code

### PR DESCRIPTION
**Pull request by sol**
I want to use `Network.HTTP.Client.MultipartFormData.renderParts` in pure code.  So what I did is generalize `renderParts` over arbitrary applicative functors.  This allows me to use it with `runIdentity`.

@snoyberg please feel free to bikeshed, anything that enables my use case works for me.

### CODE REVIEW

**Comment by snoyberg** [comment]
My only real concern is that this is a breaking change... but it seems to be warranted in this case. Can you add a major version bump and a note in the ChangeLog?

File: http-client/ChangeLog.md [hunk 1]
```diff
@@ -1,5 +1,11 @@
 # Changelog for http-client
 
+## 0.6.0
+
+* Generalize `renderParts` over arbitrary applicative functors.  One particular
+  use case that is enabled by this change is than now `renderParts` can be used
```
Comment (snoyberg) [inline http-client/ChangeLog.md]:
```suggestion
  use case that is enabled by this change is that now `renderParts` can be used
```

**Comment by sol** [comment]
@snoyberg thanks for the quick review.

> My  only real concern is that this is a breaking change... but it seems to  be warranted in this case.

This is true.

However, breakage should be minimal.  Specifically I went with

```haskell
type Part = PartM IO

data PartM m = Part ..
```

instead of (what I would usually prefer)

```haskell
type IOPart = Part IO

data Part m = Part ...
```
to minimize breakage.

> Can you add a major version bump and a note  in the ChangeLog?

Done.

**Comment by snoyberg** [comment]
I like the approach, thank you :). I put out a tweet mentioning this:

https://twitter.com/snoyberg/status/1082688452133601282

Unless I get someone screaming at me in the next 12 hours, I'll merge and release this tomorrow.

File: http-client/ChangeLog.md (hunk 1 above)
Comment (sol) [reply http-client/ChangeLog.md]: fixed.

**Comment by sol** [comment]
@snoyberg thanks a lot!

**Pull request merged**